src/currentprocess/terminalsource.rs

```diff
@@ -73,14 +74,17 @@ enum TerminalInnerLocked {
 }
 
 impl ColorableTerminal {
-    /// Construct a terminal for the selected stream. Colors written to the
-    /// terminal will be discarded if the stream is not a tty.
+    /// Construct a terminal for the selected stream.
+    /// Colors written to the terminal will be discarded if the stream is not a tty,
+    /// unless the `RUSTUP_TERM_COLOR` override environment variable is set.
     pub(super) fn new(stream: StreamSelector) -> Self {
-        let is_a_tty = stream.is_a_tty();
-        let choice = if is_a_tty {
-            ColorChoice::Auto
-        } else {
-            ColorChoice::Never
+        let env_override = process().var("RUSTUP_TERM_COLOR").ok();
+        let choice = match env_override.as_deref() {
+            Some("always") => ColorChoice::Always,
+            Some("auto") => ColorChoice::Auto,
```

Comment:
this choice is not needed: it is the same as `_ if stream.is_a_tty` by intent. I suggest just eliding it, so there will be:
Some(always)
Some(never)
_ if stream.is_a_tty
_ => ColorChoice::Never

Reply:
I agree. I also removed the `.ok()` part to match on `Ok` instead.